### PR TITLE
Ldap configuration

### PR DESCRIPTION
**Pull request by bojmil-armedia**
Waiting for the final version of ldap.yaml file template.

### CODE REVIEW

File: vagrant/provisioning/roles/arkcase-ldap-config/tasks/main.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@
     - "/home/arkcase/.arkcase/acm/spring/spring-config-{{ old_config_id }}-ldap.properties"
     - "/home/arkcase/.arkcase/acm/spring/spring-config-{{ old_config_id }}-ldap.xml"
 
-- name: ArkCase LDAP configuration
+- name: ArkCase LDAP configuration (v3.3.6 and bellow)
```
Comment (david-oc-miller) [inline vagrant/provisioning/roles/arkcase-ldap-config/tasks/main.yml]: typo ... "below"

File: vagrant/provisioning/roles/arkcase-ldap-config/tasks/main.yml [hunk 2]
```diff
@@ -30,4 +30,22 @@
...
+      dest: "/home/arkcase/.arkcase/acm/spring/spring-config-user-{{ new_config_id }}-ad.xml"
+  when: arkcase_version != "" and arkcase_version is version('3.3.7', '<')
+
+- name: ensure config server ldap folder exists
+  become: yes
+  become_user: arkcase
+  file:
+    path: /home/arkcase/.arkcase/acm/acm-config-server-repo/ldap
+    state: directory
+  when: arkcase_version != "" and arkcase_version is version('3.3.7', '>=')
```
Comment (david-oc-miller) [inline vagrant/provisioning/roles/arkcase-ldap-config/tasks/main.yml]: Probably should handle the case when `arkcase_version == ""`, like when deploying the current snapshot.

Comment (bojmil-armedia) [reply vagrant/provisioning/roles/arkcase-ldap-config/tasks/main.yml]: Thanks for noticing this

**Comment by bojmil-armedia** [comment]
@dzdimov @david-oc-miller Please review. I want to merge the LDAP configuration changes as soon as possible.

**Review by dzdimov — APPROVED** [review]

File: vagrant/provisioning/roles/arkcase-ldap-config/templates/ldap.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
+    {{ ldap_user_domain_bean_name }}:
+      base: {{ ldap_base }}
+      authUserDn: {{ ldap_bind_user }}
+      authUserPassword: ENC({{ encrypted_ldap_bind_password }})
+      userSearchBase: {{ ldap_user_no_base }}
+      groupSearchBase: {{ ldap_group_no_base }}
+      ldapUrl: {{ ldap_url }}
+      userAttributes: cn,userAccountControl,sn,givenName,dn,distinguishedname,memberOf,sAMAccountName,userPrincipalName,uid,uidNumber,c,co,company,department,title,shadowMax,shadowLastChange,msDS-UserPasswordExpiryTimeComputed,modifyTimestamp,description,member
```
Comment (david-oc-miller) [inline vagrant/provisioning/roles/arkcase-ldap-config/templates/ldap.yaml:8]: Out of curiosity, which authentication class is the hydrated Spring bean using?

Comment (bojmil-armedia) [reply vagrant/provisioning/roles/arkcase-ldap-config/templates/ldap.yaml:8]: AcmLdapSyncConfig holds the properties values and a bean is created at runtime with a name ldap_user_domain_bean_name + "_sync"

**Pull request merged**